adapters/avail/src/service.rs

```diff
@@ -0,0 +1,214 @@
+
+    type Error = anyhow::Error;
+
+    // Make an RPC call to the node to get the finalized block at the given height, if one exists.
+    // If no such block exists, block until one does.
+    async fn get_finalized_at(&self, height: u64) -> Result<Self::FilteredBlock, Self::Error> {
+        let node_client = self.node_client.clone();
+        let confidence_url = self.confidence_url(height);
+        let appdata_url = self.appdata_url(height);
+
+        wait_for_confidence(&confidence_url).await?;
+        let appdata = wait_for_appdata(&appdata_url, height as u32).await?;
+        info!("Appdata: {:?}", appdata);
+
+        let hash = node_client
+            .rpc()
+            .block_hash(Some(height.into()))
+            .await?
+            .unwrap();
+
+        let header = node_client.rpc().header(Some(hash)).await?.unwrap();
+
+        let header = AvailHeader::new(header, hash);
+        let transactions = appdata
+            .extrinsics
+            .iter()
+            .map(AvailBlobTransaction::new)
+            .collect();
+        Ok(AvailBlock {
+            header,
+            transactions,
+        })
+    }
+
+    // Make an RPC call to the node to get the block at the given height
+    // If no such block exists, block until one does.
+    async fn get_block_at(&self, height: u64) -> Result<Self::FilteredBlock, Self::Error> {
+        self.get_finalized_at(height).await
+    }
+
+    // Extract the blob transactions relevant to a particular rollup from a block.
+    // NOTE: The avail light client is expected to be run in app specific mode, and hence the
+    // transactions in the block are already filtered and retrieved by light client.
+    fn extract_relevant_txs(
+        &self,
+        block: &Self::FilteredBlock,
+    ) -> Vec<<Self::Spec as DaSpec>::BlobTransaction> {
+        block.transactions.clone()
+    }
+
+    // Extract the inclusion and completenss proof for filtered block provided.
+    // The output of this method will be passed to the verifier.
+    // NOTE: The light client here has already completed DA sampling and verification of inclusion and soundness.
+    async fn get_extraction_proof(
+        &self,
+        _block: &Self::FilteredBlock,
+        _blobs: &[<Self::Spec as DaSpec>::BlobTransaction],
+    ) -> (
+        <Self::Spec as DaSpec>::InclusionMultiProof,
+        <Self::Spec as DaSpec>::CompletenessProof,
+    ) {
+        ((), ())
+    }
+
+    async fn send_transaction(&self, blob: &[u8]) -> Result<(), Self::Error> {
+        let data_transfer = api::tx()
+            .data_availability()
+            .submit_data(BoundedVec(blob.to_vec()));
+
+        let extrinsic_params = AvailExtrinsicParams::new_with_app_id(7.into());
+
+        let h = self
+            .node_client
+            .tx()
+            .sign_and_submit_then_watch(&data_transfer, &self.signer, extrinsic_params)
+            .await?;
+
+        println!("Transaction submitted: {:#?}", h.extrinsic_hash());
```

Comment:
What about `tracing::info` instead of `println`

Reply:
Fixed by #980